src/transports/http.c

```diff
@@ -246,7 +246,7 @@ static int parse_authenticate_response(
 
 	git_vector_foreach(www_authenticate, i, challenge) {
 		if (auth_context_match(&context, t, challenge_match, challenge) < 0)
-			return -1;
+			continue;
```

Comment:
🤔

One problem here is that it seems that we have overloaded the return value of `auth_context_match`, and it's returning `-1` when there was no scheme match.  I think that might be the root cause of the problems here.

I think that we should change `auth_context_match` to return `0` (and leave `context` unset) when there's no match, instead of [returning `-1`](https://github.com/libgit2/libgit2/blob/453ed9e41eaa321321701409dd908737ff0b28bf/src/transports/http.c#L145).  I think that this bit can then remain unchanged - on `auth_context_match` returning `-1` on error, it will propagate this error, and returning `0` without setting `context`, it will continue.

Reply:
Yes, this might be a better idea.

The only other usage is by ``apply_credentials`` and I'm not sure if this needs to do something more to handle the situation where return value is 0 but the output context is NULL?

 https://github.com/libgit2/libgit2/blob/a8d447f68076d1520f69649bb52629941be7031f/src/transports/http.c#L169-L192